lib/inspec/plugin/v2/installer.rb

```diff
@@ -368,9 +377,19 @@ def install_gem_to_plugins_dir(new_plugin_dependency, # rubocop: disable Metrics
       # of the same gem on the stack or a malformed dependency.
       begin
         solution.each do |activation_request|
-          unless activation_request.full_spec.activated?
-            activation_request.full_spec.activate
+          requested_gemspec = activation_request.full_spec
+          next if requested_gemspec.activated?
+
+          gem_name = requested_gemspec.name
+          loaded_gem = Gem.loaded_specs[gem_name]
+          if loaded_gem
+            if requested_gemspec.version > loaded_gem.version
```

Comment:
I am having confusion understanding how does `requested_gemspec.version` and `loaded_gem.version` will be different? Because loaded gem is using the same spec of requested_spec...How would it load a different spec version internally? I know in this scenarios 2 specs exists in the system.. but which one does `Gem.loaded_specs[gem_name]` picks?

Reply:
No both are different.. loaded gemspec is loaded via bundler when Inspec is booted while requested gemspec is loaded via Gem::RequesSet object